e2e-test-server/src/main/java/com/google/cloud/opentelemetry/endtoend/ScenarioHandlerManager.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.endtoend;
+
+import com.google.cloud.opentelemetry.trace.TraceConfiguration;
+import com.google.cloud.opentelemetry.trace.TraceExporter;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * A container for all scenarios we handle.
+ *
+ * <p>We "dependency inject" scenarios into the constructor.
+ */
+public class ScenarioHandlerManager {
+  private Map<String, ScenarioHandler> scenarios = new HashMap<>();
+
+  public ScenarioHandlerManager() {
+    register("/health", this::health);
+    register("/basicTrace", this::basicTrace);
+  }
+
+  /** Health check test. */
+  private Response health(Request request) {
+    return Response.ok();
+  }
+
+  /** Basic trace test. */
+  private Response basicTrace(Request request) {
+    return withTemporaryTracer(
+        (tracer) -> {
+          Span span =
+              tracer
+                  .spanBuilder("basicTrace")
+                  .setAttribute(Constants.TEST_ID, request.testId())
+                  .startSpan();
+          try {
+            return Response.ok();
+          } finally {
```

Comment:
What's the point of this `try...finally`? It seems like there's no way for `Response.ok()` to throw. Is it there to establish a consistent pattern for other tests to follow?

Reply:
Two (maytbe silly?) reasons:

1. To ensure modifications follow this pattern.
2. The bytecode should instantiate response before ending the span.